src/main/java/com/management/HumanResources/dao/FirebaseDao.java

```diff
@@ -0,0 +1,57 @@
+package com.management.HumanResources.dao;
+
+import java.util.List;
+
+import javax.annotation.PostConstruct;
+
+import com.management.HumanResources.model.Employee;
+import com.management.HumanResources.model.TimeOff;
+
+import org.springframework.stereotype.Repository;
+import org.springframework.web.reactive.function.client.WebClient;
+
+@Repository
+public class FirebaseDao extends Dao {
+  @PostConstruct
+  private void init() {
+    webClient =
+        WebClient.builder()
+            .baseUrl("TBD")
+            .defaultHeaders(
+                httpHeaders ->
+                    httpHeaders.setBasicAuth(
+                        "TODO databaseUsername", "TODO databasePassword"))
+            .build();
+  }
+
+  //Database paths are just placeholders until the actual ones are created
+
+  public List<Employee> getAllEmployees() {
+      return getListOfObjects(Employee.class, "path");
+  }
+
+  public Employee getSingleEmployee(int id) {
+      return getSingleObject(Employee.class, "path/{employeeId}", id);
+  }
+
+  public String updateSingleEmployee(Employee employee) {
```

Comment:
same here and the rest of the relevant methods in this file

Reply:
Good Suggestions 👍